community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStore.java

```diff
@@ -152,6 +157,19 @@ public static boolean isStorePresent( FileSystemAbstraction fs, File storeDir )
     private final int relGrabSize;
     private final CappedOperation<Void> transactionCloseWaitLogger;
 
+    private final ThreadLocal<AtomicReference<StoreStatement>> storeStatement = new ThreadLocal<AtomicReference<StoreStatement>>()
+    {
+        @Override
+        protected AtomicReference<StoreStatement> initialValue()
+        {
+            AtomicReference<StoreStatement> storeStatementAtomicReference = new AtomicReference<>(
+                    new StoreStatement( NeoStore.this ) );
+            statements.add(storeStatementAtomicReference);
+            return storeStatementAtomicReference;
+        }
+    };
+    private final Queue<AtomicReference> statements = new ConcurrentLinkedDeque<>(  );
+
```

Comment:
I am unsure if this should be part of the `NeoStore`, why not having a new class for this managed by the `LifeCycle`?